packages/aptos-wallet-adapter/src/WalletAdapters/Coin98Wallet.ts

```diff
@@ -0,0 +1,310 @@
+import { Types } from 'aptos';
+import {
+  WalletAccountChangeError,
+  WalletDisconnectionError,
+  WalletGetNetworkError,
+  WalletNetworkChangeError,
+  WalletNotConnectedError,
+  WalletNotReadyError,
+  WalletSignAndSubmitMessageError,
+  WalletSignMessageError,
+  WalletSignTransactionError
+} from '../WalletProviders/errors';
+import {
+  AccountKeys,
+  BaseWalletAdapter,
+  NetworkInfo,
+  scopePollingDetectionStrategy,
+  SignMessagePayload,
+  SignMessageResponse,
+  WalletAdapterNetwork,
+  WalletName,
+  WalletReadyState
+} from './BaseAdapter';
+
+interface IApotsErrorResult {
+  code: number;
+  message: string;
+}
+
+type AddressInfo = { address: string; publicKey: string };
+
+interface IConnectOptions {
+  network: string;
+}
+
+interface ICoin98Wallet {
+  connect: (options?: IConnectOptions) => Promise<AddressInfo>;
+  account: () => Promise<AddressInfo>;
+  isConnected: () => Promise<boolean>;
+  signAndSubmitTransaction(
+    transaction: any,
+    options?: any
+  ): Promise<{ hash: Types.HexEncodedBytes } | IApotsErrorResult>;
+  signTransaction(transaction: any, options?: any): Promise<Uint8Array | IApotsErrorResult>;
+  signMessage(message: SignMessagePayload): Promise<SignMessageResponse>;
+  disconnect(): Promise<void>;
+  on: (eventName: string, callback: Function) => void;
+}
+
+interface AptosWindow extends Window {
+  coin98?: {
+    aptos: ICoin98Wallet;
+  };
+}
+
+declare const window: AptosWindow;
+
+export const Coin98WalletName = 'Coin98' as WalletName<'Coin98'>;
+
+export interface Coin98WalletAdapterConfig {
+  provider?: ICoin98Wallet;
+  network?: WalletAdapterNetwork;
+  timeout?: number;
+}
+
+export class Coin98WalletAdapter extends BaseWalletAdapter {
+  name = Coin98WalletName;
+
+  url = 'https://chrome.google.com/webstore/detail/coin98-wallet/aeachknmefphepccionboohckonoeemg';
+
+  icon = 'https://coin98.s3.ap-southeast-1.amazonaws.com/Coin/c98wallet.png';
+
+  protected _provider: ICoin98Wallet | undefined;
+
+  protected _network: WalletAdapterNetwork;
+
+  protected _chainId: string;
+
+  protected _api: string;
+
+  protected _timeout: number;
+
+  protected _readyState: WalletReadyState =
+    typeof window === 'undefined' || typeof document === 'undefined'
+      ? WalletReadyState.Unsupported
+      : WalletReadyState.NotDetected;
+
+  protected _connecting: boolean;
+
+  protected _wallet: any | null;
+
+  constructor({
+    // provider,
+    network = WalletAdapterNetwork.Mainnet,
+    timeout = 10000
+  }: Coin98WalletAdapterConfig = {}) {
+    super();
+
+    this._provider = typeof window !== 'undefined' ? window.coin98?.aptos : undefined;
+    this._network = network;
```

Comment:
I think it is better to align with other wallets and left out the default network by setting it to undefined

Reply:
Commented fixed. Thanks for your review